File: Offerte.Ui/src/models/agente.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('agente')
export class Agente extends Entity {

    @field()
    id: number;
    @field()
    ragioneSociale: string;
    @field()
    indirizzo: string;
    @field()
    cap: string;
    @field()
    localita: string;
    @field()
    codiceFiscale: string;
    @field()
    partitaIVA: string;
    @field()
    telefono: string;
    @field()
    numeroFax: string;
    @field()
    budget: number;
    @field()
    fattProg: number;
    @field()
    agenteOmega: number;
    @field()
    name: string;
    @field()
    surname: string;
    @field()
    password: string;
    @field()
    supervisore: boolean;

    role: string
    token: string
    disabled: boolean
    
    constructor(data?: any) {
        super()
        super.init(this, data);      
    }

    static newItem(): Agente {
        let item = new Agente();
        return item;
    }
}

@resource('agente')
export class AgenteSearchModel extends Entity implements Pagination {
    
    @field()
    ragioneSociale?: string;
    @field()
    indirizzo?: string;
    @field()
    cap?: string;
    @field()
    localita?: string;
    @field()
    codiceFiscale?: string;
    @field()
    partitaIVA?: string;
    @field()
    telefono?: string;
    @field()
    numeroFax?: string;
    @field()
    budget?: number;
    @field()
    fattProg?: number;
    @field()
    agenteOmega?: number;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}

File: Offerte.Core/Models/Agente.cs
using System;
using System.Collections.Generic;

namespace Offerte.Core.Models;

public partial class Agente
{
    public long Id { get; set; }

    public string? RagioneSociale { get; set; }

    public string? Indirizzo { get; set; }

    public string? Cap { get; set; }

    public string? Localita { get; set; }

    public string? CodiceFiscale { get; set; }

    public string? PartitaIva { get; set; }

    public string? Telefono { get; set; }

    public string? NumeroFax { get; set; }

    public double? Budget { get; set; }

    public double? FattProg { get; set; }

    public string? Password { get; set; }

    public bool Supervisore { get; set; }

    public string? Mail { get; set; }

    public bool FarEast { get; set; }

    public string? PasswordMail { get; set; }

    public string? Name { get; set; }

    public string? Surname { get; set; }

    public virtual ICollection<AgenteGruppo> AgenteGruppo { get; set; } = new List<AgenteGruppo>();

    public virtual AgentePassword? AgentePassword { get; set; }

    public virtual ICollection<Offerta> Offerta { get; set; } = new List<Offerta>();
}
